### ISSUE TITLE
Package not uninstallable

### ISSUE THREAD

**Issue by milkboy** [robinostlund/volkswagencarnet#148]
Due to the files being installed into the root of site-packages, it's not possible do do `pip uninstall volkswagencarnet`, as it would try to remove everything under site-packages.

Would it make sense to fix it so that the files are installed in a proper location instead? It's a breaking change though, since all imports in dependent projects need to be changed...

Having the files in the proper location would also speed up development/debugging for the HA integrations, as the folder can be bind mounted directly in the development environment.

I have zero experience in actually packaging any python projects, but could try to have a look ~~when~~ _if_ I have some spare time

**Comment by robinostlund** [comment]
@JohNan if i dont remember wrong, you tried this once right?

**Comment by milkboy** [comment]
```
 $ pip uninstall volkswagencarnet
Found existing installation: volkswagencarnet 4.4.38
Uninstalling volkswagencarnet-4.4.38:
  Would remove:
    /home/michael/IdeaProjects/venv/lib/python3.9/site-packages/*
  Would not remove (might be manually added):

```

**Comment by robinostlund** [comment]
Think we should start with changing the folderstructure as https://github.com/lendy007/skodaconnect. Think that could be one potential issue, but it may be causing some fun breaks :D

**Comment by milkboy** [comment]
yea. The files need to be moved at least. Should not be a biggie though.. I'll try something soonish

**Comment by milkboy** [comment]
```
$ python -m build
* Creating venv isolated environment...
* Installing packages in isolated environment... (pytest>=5, setuptools>=42, setuptools_scm, wheel)
* Getting dependencies for sdist...
running egg_info
writing volkswagencarnet.egg-info/PKG-INFO
writing dependency_links to volkswagencarnet.egg-info/dependency_links.txt
writing top-level names to volkswagencarnet.egg-info/top_level.txt
adding license file 'LICENSE.txt'
* Building sdist...
running sdist
running egg_info
writing volkswagencarnet.egg-info/PKG-INFO
writing dependency_links to volkswagencarnet.egg-info/dependency_links.txt
writing top-level names to volkswagencarnet.egg-info/top_level.txt
adding license file 'LICENSE.txt'
running check
copying files to volkswagencarnet-1.0.0...
Writing volkswagencarnet-1.0.0/setup.cfg
* Building wheel from sdist
* Creating venv isolated environment...
* Installing packages in isolated environment... (pytest>=5, setuptools>=42, setuptools_scm, wheel)
* Getting dependencies for wheel...
running egg_info
writing volkswagencarnet.egg-info/PKG-INFO
writing dependency_links to volkswagencarnet.egg-info/dependency_links.txt
writing top-level names to volkswagencarnet.egg-info/top_level.txt
listing git files failed - pretending there aren't any
reading manifest file 'volkswagencarnet.egg-info/SOURCES.txt'
adding license file 'LICENSE.txt'
* Installing packages in isolated environment... (wheel)
* Building wheel...
running bdist_wheel
running build
running build_py
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
copying build/lib/volkswagencarnet/vw_dashboard.py -> build/bdist.linux-x86_64/wheel/volkswagencarnet
copying build/lib/volkswagencarnet/vw_const.py -> build/bdist.linux-x86_64/wheel/volkswagencarnet
copying build/lib/volkswagencarnet/vw_vehicle.py -> build/bdist.linux-x86_64/wheel/volkswagencarnet
copying build/lib/volkswagencarnet/__init__.py -> build/bdist.linux-x86_64/wheel/volkswagencarnet
copying build/lib/volkswagencarnet/vw_connection.py -> build/bdist.linux-x86_64/wheel/volkswagencarnet
copying build/lib/volkswagencarnet/vw_utilities.py -> build/bdist.linux-x86_64/wheel/volkswagencarnet
copying build/lib/tests/dummy_test.py -> build/bdist.linux-x86_64/wheel/tests
copying build/lib/tests/__init__.py -> build/bdist.linux-x86_64/wheel/tests
copying build/lib/tests/integration_test.py -> build/bdist.linux-x86_64/wheel/tests
copying build/lib/tests/credentials.py -> build/bdist.linux-x86_64/wheel/tests
running install_egg_info
running egg_info
writing volkswagencarnet.egg-info/PKG-INFO
writing dependency_links to volkswagencarnet.egg-info/dependency_links.txt
writing top-level names to volkswagencarnet.egg-info/top_level.txt
listing git files failed - pretending there aren't any
reading manifest file 'volkswagencarnet.egg-info/SOURCES.txt'
adding license file 'LICENSE.txt'
running install_scripts
adding license file "LICENSE.txt" (matched pattern "LICEN[CS]E*")
creating build/bdist.linux-x86_64/wheel/volkswagencarnet-1.0.0.dist-info/WHEEL
creating '<snip>/dist/tmp3iuqv9sz/volkswagencarnet-1.0.0-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'tests/__init__.py'
adding 'tests/credentials.py'
adding 'tests/dummy_test.py'
adding 'tests/integration_test.py'
adding 'volkswagencarnet/__init__.py'
adding 'volkswagencarnet/vw_connection.py'
adding 'volkswagencarnet/vw_const.py'
adding 'volkswagencarnet/vw_dashboard.py'
adding 'volkswagencarnet/vw_utilities.py'
adding 'volkswagencarnet/vw_vehicle.py'
adding 'volkswagencarnet-1.0.0.dist-info/LICENSE.txt'
adding 'volkswagencarnet-1.0.0.dist-info/METADATA'
adding 'volkswagencarnet-1.0.0.dist-info/WHEEL'
adding 'volkswagencarnet-1.0.0.dist-info/top_level.txt'
adding 'volkswagencarnet-1.0.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Successfully built volkswagencarnet-1.0.0.tar.gz and volkswagencarnet-1.0.0-py3-none-any.whl
```

```
$ pip install dist/volkswagencarnet-1.0.0.tar.gz 
Processing ./dist/volkswagencarnet-1.0.0.tar.gz
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Building wheels for collected packages: volkswagencarnet
  Building wheel for volkswagencarnet (PEP 517) ... done
  Created wheel for volkswagencarnet: filename=volkswagencarnet-1.0.0-py3-none-any.whl size=36016 sha256=4c009d4096e3dfd2a963a9211ff94182d35287531b00287249dd7563f00187ae
  Stored in directory: <snip>/.cache/pip/wheels/16/6e/37/547bd29be437d2338c0f292d030372055ab6fd879d8f42778e
Successfully built volkswagencarnet
Installing collected packages: volkswagencarnet
Successfully installed volkswagencarnet-1.0.0
```

okay.. tested and some minor changes required in the HA integration, but it works

**Comment by milkboy** [comment]
Something like #149

**Comment by milkboy** [comment]
Files are then actually installed in `site-packages/volkswagencarnet/`

**Comment by milkboy** [comment]
https://github.com/robinostlund/homeassistant-volkswagencarnet/pull/366 I think this was the minimal change set for the HA component, so not too bad

**Comment by robinostlund** [comment]
Seems to work as it should now :)
```
pip3 uninstall volkswagencarnet
Found existing installation: volkswagencarnet 4.4.44
Uninstalling volkswagencarnet-4.4.44:
  Would remove:
    /Users/robin/.venv/python3/lib/python3.9/site-packages/tests/*
    /Users/robin/.venv/python3/lib/python3.9/site-packages/volkswagencarnet-4.4.44.dist-info/*
    /Users/robin/.venv/python3/lib/python3.9/site-packages/volkswagencarnet/*
Proceed (y/n)? n

ls -l /Users/robin/.venv/python3/lib/python3.9/site-packages/volkswagencarnet/
total 336
-rw-r--r--  1 robin  staff      0 Feb  7 21:00 __init__.py
drwxr-xr-x  8 robin  staff    256 Feb  7 21:00 __pycache__
-rw-r--r--  1 robin  staff  60913 Feb  7 21:00 vw_connection.py
-rw-r--r--  1 robin  staff   2022 Feb  7 21:00 vw_const.py
-rw-r--r--  1 robin  staff  26606 Feb  7 21:00 vw_dashboard.py
-rw-r--r--  1 robin  staff   2472 Feb  7 21:00 vw_utilities.py
-rw-r--r--  1 robin  staff  71974 Feb  7 21:00 vw_vehicle.py
```

**Comment by milkboy** [comment]
Crap.. except the test folder.. forgot to triple check everything.. need to tell the setup part to skip the tests when creating the package

**Comment by milkboy** [comment]
Or move the `volkswagencarnet` folder into a `src` folder, and point the root to `src` instead, not sure which is easier or cleaner

**Comment by robinostlund** [comment]
Tried to move it to a src folder and change root etc, but still it adds the tests folder. Must be something else 😅 but now time to get some rest 🥱😴

**Comment by JohNan** [comment]
> @JohNan if i dont remember wrong, you tried this once right?

Heh. Yes I think I did something similar. But we reverted it for some reason that I can't remember. It's probably in a PR somewhere. 🤔

**Comment by robinostlund** [comment]
@milkboy if you want to test, just run `python -m build --sdist --wheel --outdir dist/ .` in the root folder

**Comment by milkboy** [comment]
Fixed by #159
